packages/builder/src/package.ts

```diff
@@ -37,7 +37,7 @@ export interface PackagePublishCall {
 export class PackageReference {
   static DEFAULT_TAG = 'latest';
   static DEFAULT_PRESET = 'main';
-  static PACKAGE_REGEX = /^(?<name>@?[a-z0-9][A-Za-z0-9-]{1,30}[a-z0-9])(?::(?<version>[^@]+))?(@(?<preset>[^\s]+))?$/;
+  static PACKAGE_REGEX = /^(?<name>@?[a-z0-9][A-Za-z0-9-]{1,}[a-z0-9])(?::(?<version>[^@]+))?(@(?<preset>[^\s]+))?$/;
```

Comment:
The `PackageReference.isValid` fn now only uses the regex to validate, not the constructor, so it would not return `false` when given a too long package name.

Reply:
Yeah im only using the isValid function to check whether it matches the format, however when setting the length inside the regex it does fail on `.match` due to length

Reply:
Agree on the `parse` functionality with prettier errors, but the `PackageReference.isValid` should return `false` when it is too long, and with this change it doesn't. Maybe instead of validating with the regex inside the `isValid` you can use `.parse` with a try catch.